orthopairs/src/main/java/org/reactome/release/orthopairs/OrthopairFileGenerator.java

```diff
@@ -0,0 +1,93 @@
+package org.reactome.release.orthopairs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.simple.JSONObject;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+
+public class OrthopairFileGenerator {
+    private static final Logger logger = LogManager.getLogger();
+    // Create source-target protein mapping file
+    public static void createProteinHomologyFile(String speciesKey, String sourceTargetProteinMappingFilename, JSONObject speciesJSON, Map<String,Set<String>> speciesProteinHomologs ) throws IOException {
+
+        logger.info("\tGenerating " + sourceTargetProteinMappingFilename);
+        File sourceTargetProteinMappingFile = new File(sourceTargetProteinMappingFilename);
+        if (sourceTargetProteinMappingFile.exists()) {
+            sourceTargetProteinMappingFile.delete();
+        }
+        sourceTargetProteinMappingFile.createNewFile();
+
+        List<String> sourceProteinIds = new ArrayList<>();
+        sourceProteinIds.addAll(speciesProteinHomologs.keySet());
+        Collections.sort(sourceProteinIds);
+        for (String sourceProteinId : sourceProteinIds) {
+            Set<String> targetProteinIds = speciesProteinHomologs.get(sourceProteinId);
+            targetProteinIds.remove("LDO");
+            String proteinOrthologLine = sourceProteinId.split("=")[1] + "\t";
+            List<String> cleanTargetProteinIds = new ArrayList<>();
+            for (String targetProteinId : targetProteinIds) {
+                cleanTargetProteinIds.add(targetProteinId.split("=")[1]);
+            }
+            Collections.sort(cleanTargetProteinIds);
```

Comment:
For transforming operations on collections, it can be more declarative to use streams (though not always) to show what's being done to the data.  The book Effective Java recommends that it's usually best to consider both ways and decide which way is clearer.  With streams, this could be done this way:

```
String targetProteinIds = speciesProteinHomologs.get(sourceProteinId)
	.stream()
	.filter(targetProteinId -> !targetProteinId.equals("LDO"))
	.map(targetProteinId -> targetProteinId.split("=")[1])
	.sorted()
	.collect(Collectors.joining(" "));
```

Reply:
Actually, this is done in both methods that create files, so a private method might be better:

```
private static String getTargetProteinsAsString(Set<String> targetProteins) {
	return targetProteins
		.stream()
		.filter(targetProteinId -> !targetProteinId.equals("LDO"))
		.map(targetProteinId -> targetProteinId.split("=")[1])
		.sorted()
		.collect(Collectors.joining(" "));
}
```